src/settings/toml/deploy_target.rs

```diff
@@ -0,0 +1,161 @@
+use crate::settings::toml::Route;
+
+impl DeployTarget {
+    pub fn build(
+        script_name: &str,
+        route_config: &RouteConfig,
+    ) -> Result<DeployTarget, failure::Error> {
+        if route_config.workers_dev_false_by_itself() || route_config.has_conflicting_targets() {
+            failure::bail!("you must set workers_dev = true or provide a zone_id and route/routes.")
+        }
+
+        if route_config.is_zoneless() {
+            DeployTarget::build_zoneless(script_name, route_config)
+        } else if route_config.is_zoned() {
+            DeployTarget::build_zoned(script_name, route_config)
+        } else {
+            failure::bail!("No deploy target specified");
+        }
+    }
+
+    fn build_zoneless(
+        script_name: &str,
+        route_config: &RouteConfig,
+    ) -> Result<DeployTarget, failure::Error> {
+        if let Some(account_id) = &route_config.account_id {
+            // TODO: Deserialize empty strings to None
+            if account_id.is_empty() {
+                failure::bail!("field `account_id` is required to deploy to workers.dev");
+            }
+            let zoneless = Zoneless {
+                script_name: script_name.to_string(),
+                account_id: account_id.to_string(),
+            };
+
+            Ok(DeployTarget::Zoneless(zoneless))
+        } else {
+            failure::bail!("field `account_id` is required to deploy to workers.dev");
+        }
+    }
+
+    fn build_zoned(
+        script_name: &str,
+        route_config: &RouteConfig,
+    ) -> Result<DeployTarget, failure::Error> {
+        if let Some(zone_id) = &route_config.zone_id {
+            // TODO: Deserialize empty strings to None
+            if zone_id.is_empty() {
+                failure::bail!("field `zone_id` is required to deploy to routes");
+            }
+
+            if route_config.has_conflicting_targets() {
+                failure::bail!("specify either `route` or `routes`");
+            }
+
+            let mut zoned = Zoned {
+                zone_id: zone_id.to_owned(),
+                routes: Vec::new(),
+            };
+
+            // TODO: these should be an if/else if block; write deserializer
```

Comment:
Do you still plan on getting these todos in for this PR?

Reply:
this already happened! i can remove this TODO.